frontend/src/metabase/visualizations/visualizations/LinkViz/LinkViz.tsx

```diff
@@ -1,76 +1,153 @@
+          display: entity.display,
+        },
+      },
+    });
+  };
 
-  const showEditBox = isEditing && !isPreviewing && !isSettings;
+  const [inputIsFocused, { turnOn: onFocusInput, turnOff: onBlurInput }] =
+    useToggle();
+
+  useEffect(() => {
+    // if the url was auto-filled from the entity, focus the input
+    if (previousUrl === undefined && !!url) {
+      setAutoFocus(true);
+    }
+  }, [previousUrl, url]);
+
+  if (entity) {
+    const wrappedEntity = Search.wrapEntity({
+      ...entity,
+      database_id: entity.db_id ?? entity.database_id,
+      table_id: entity.model === "table" ? entity.id : undefined,
+      collection: {},
+    });
+
+    if (isEditing) {
+      return (
+        <EditLinkCardWrapper>
+          <EntityDisplay entity={wrappedEntity} showDescription={false} />
+        </EditLinkCardWrapper>
+      );
+    }
+
+    return (
+      <DisplayLinkCardWrapper>
+        <CardLink to={wrappedEntity.getUrl()} target="_blank" rel="noreferrer">
+          <EntityDisplay entity={wrappedEntity} showDescription />
+        </CardLink>
+      </DisplayLinkCardWrapper>
+    );
+  }
 
-  if (showEditBox) {
+  if (isEditing) {
     return (
       <EditLinkCardWrapper>
-        <Input
-          autoFocus={isNew}
-          placeholder={"https://example.com"}
-          value={url ?? ""}
-          onChange={handleLinkChange}
-          fullWidth
-          // the dashcard really wants to turn all mouse events into drag events
-          onMouseDown={e => e.stopPropagation()}
-        />
+        <TippyPopover
+          visible={!!url?.length && inputIsFocused && !isUrlString(url)}
+          content={
+            <SearchResultsContainer>
+              <SearchResults
+                searchText={url?.trim()}
+                onEntitySelect={handleEntitySelect}
+              />
+            </SearchResultsContainer>
+          }
+          placement="bottom"
+        >
+          <Input
+            fullWidth
+            value={url ?? ""}
+            autoFocus={autoFocus}
+            placeholder={t`https://example.com`}
+            onChange={e => handleLinkChange(e.target.value)}
+            onFocus={onFocusInput}
+            onBlur={onBlurInput}
+            // the dashcard really wants to turn all mouse events into drag events
+            onMouseDown={e => e.stopPropagation()}
+          />
+        </TippyPopover>
       </EditLinkCardWrapper>
     );
   }
 
-  const displayIcon = isEmpty(url) ? "question" : "link";
+  const urlIcon = isEmpty(url) ? "question" : "link";
 
   return (
     <DisplayLinkCardWrapper>
-      <CardLink to={url ?? ""} target="_blank" rel="noreferrer">
-        <Icon name={displayIcon} />
-        <Ellipsified style={{ minWidth: 0 }}>
-          {isEmpty(url) ? t`Choose a link` : url}
-        </Ellipsified>
+      <CardLink to={url ? url : ""} target="_blank" rel="noreferrer">
+        <Icon name={urlIcon} ml={1} mr={1} color={color("brand")} />
```

Comment:
Could we use emotion here please?

Reply:
I guess this is my biggest beef with emotion, and where something like tailwind would be really useful. Going to a separate file to add a little margin around an icon is pretty annoying, and probably of very little value in terms of maintainability.